### ISSUE TITLE
Error: unable to build kubernetes objects from release manifest

### ISSUE THREAD

**Issue by hannius** [traefik/traefik-helm-chart#476]
### Welcome!

- [X] Yes, I've searched similar issues on [GitHub](https://github.com/traefik/traefik-helm-chart/issues) and didn't find any.
- [X] Yes, I've searched similar issues on the [Traefik community forum](https://community.traefik.io) and didn't find any.

### What version of the Traefik's Helm Chart are you using?

3.4.2

### What version of Traefik are you using?

10.1.0 

### What did you do?

Hello ,when i use the traefik-helm-chart(10.1.0), i encountered the following problems: 
```
➜  traefik helm install traefik-test --namespace kube-system -f values.yaml --debug --dry-run .
install.go:172: [debug] Original chart version: ""
install.go:189: [debug] CHART PATH: /home/leiyawu/repo/traefik

install.go:189: [debug] WARNING: This chart or one of its subcharts contains CRDs. Rendering may fail or contain inaccuracies.
Error: unable to build kubernetes objects from release manifest: unable to decode "": resource.metadataOnlyObject.ObjectMeta: v1.ObjectMeta.Annotations: ReadString: expects " or n, but found 9, error found in #10 byte of ...|io/port":9100,"prome|..., bigger context ...|metheus.io/path":"/metrics","prometheus.io/port":9100,"prometheus.io/scrape":"true"},"labels":{"app.|...
helm.go:81: [debug] unable to decode "": resource.metadataOnlyObject.ObjectMeta: v1.ObjectMeta.Annotations: ReadString: expects " or n, but found 9, error found in #10 byte of ...|io/port":9100,"prome|..., bigger context ...|metheus.io/path":"/metrics","prometheus.io/port":9100,"prometheus.io/scrape":"true"},"labels":{"app.|...
unable to build kubernetes objects from release manifest
helm.sh/helm/v3/pkg/action.(*Install).Run
        /home/circleci/helm.sh/helm/pkg/action/install.go:257
main.runInstall
        /home/circleci/helm.sh/helm/cmd/helm/install.go:241
main.newInstallCmd.func2
        /home/circleci/helm.sh/helm/cmd/helm/install.go:120
github.com/spf13/cobra.(*Command).execute
        /go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:842
github.com/spf13/cobra.(*Command).ExecuteC
        /go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:950
github.com/spf13/cobra.(*Command).Execute
        /go/pkg/mod/github.com/spf13/cobra@v1.0.0/command.go:887
main.main
        /home/circleci/helm.sh/helm/cmd/helm/helm.go:80
runtime.main
        /usr/local/go/src/runtime/proc.go:203
runtime.goexit
        /usr/local/go/src/runtime/asm_amd64.s:1373
```
The helm and traefik version is below:
```
➜  traefik helm repo ls
NAME            URL                                              
traefik         https://helm.traefik.io/traefik                  
traefik-github  https://containous.github.io/traefik-helm-chart  
elastic         https://helm.elastic.co                          
kong            https://charts.konghq.com                        
stable          https://charts.helm.sh/stable                    
rancher-latest  https://releases.rancher.com/server-charts/latest
➜  traefik helm pull traefik/traefik --version v10.1.10
```

pls help to solve it,tks

### What did you see instead?

```
Error: unable to build kubernetes objects from release manifest: unable to decode "": resource.metadataOnlyObject.ObjectMeta: v1.ObjectMeta.Annotations: ReadString: expects " or n, but found 9, error found in #10 byte of ...|io/port":9100,"prome|..., bigger context ...|metheus.io/path":"/metrics","prometheus.io/port":9100,"prometheus.io/scrape":"true"},"labels":{"app.|...
helm.go:81: [debug] unable to decode "": resource.metadataOnlyObject.ObjectMeta: v1.ObjectMeta.Annotations: ReadString: expects " or n, but found 9, error found in #10 byte of ...|io/port":9100,"prome|..., bigger context ...|metheus.io/path":"/metrics","prometheus.io/port":9100,"prometheus.io/scrape":"true"},"labels":{"app.|...
unable to build kubernetes objects from release manifest
```

### What is your environment & configuration?

```
➜  traefik helm version
version.BuildInfo{Version:"v3.4.2", GitCommit:"23dd3af5e19a02d4f4baa5b2f242645a1a3af629", GitTreeState:"clean", GoVersion:"go1.14.13"}
```
```
  traefik helm pull traefik/traefik --version v10.1.0
```

**Comment by faust64** [comment]
Hi,

This issue is already fixed:

```
$ git diff v10.1.0..v10.1.2 traefik/templates/deployment.yaml
diff --git a/traefik/templates/deployment.yaml b/traefik/templates/deployment.yaml
index 5d5d773..59102ab 100644
--- a/traefik/templates/deployment.yaml
+++ b/traefik/templates/deployment.yaml
@@ -33,7 +33,7 @@ metadata:
   {{- if .Values.metrics.prometheus }}
     prometheus.io/scrape: "true"
     prometheus.io/path: "/metrics"
-    prometheus.io/port: {{ (index .Values.ports .Values.metrics.prometheus.entryPoint).port }}
+    prometheus.io/port: {{ quote (index .Values.ports .Values.metrics.prometheus.entryPoint).port }}
   {{- end }}
   {{- end }}
 spec:
```

Please try with the last version of Traefik Helm Chart.

**Comment by hannius** [comment]
tks, i updated to version 10.1.1,the problem have been resolved.谢谢

**Comment by jakubhajek** [comment]
The issue has been fixed in PR #456. Closing it.
